[2025-09-16T15:24:18.714+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_bronze_load scheduled__2025-09-16T15:05:00+00:00 [queued]>
[2025-09-16T15:24:18.717+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_bronze_load scheduled__2025-09-16T15:05:00+00:00 [queued]>
[2025-09-16T15:24:18.717+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-09-16T15:24:18.721+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): run_bronze_load> on 2025-09-16 15:05:00+00:00
[2025-09-16T15:24:18.724+0000] {standard_task_runner.py:60} INFO - Started process 9302 to run task
[2025-09-16T15:24:18.725+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'opensky_data_pipeline_dag', 'run_bronze_load', 'scheduled__2025-09-16T15:05:00+00:00', '--job-id', '102', '--raw', '--subdir', 'DAGS_FOLDER/opensky_data_pipeline_dag.py', '--cfg-path', '/tmp/tmpoft2sdly']
[2025-09-16T15:24:18.726+0000] {standard_task_runner.py:88} INFO - Job 102: Subtask run_bronze_load
[2025-09-16T15:24:18.748+0000] {task_command.py:423} INFO - Running <TaskInstance: opensky_data_pipeline_dag.run_bronze_load scheduled__2025-09-16T15:05:00+00:00 [running]> on host 64220da1adfe
[2025-09-16T15:24:18.872+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='opensky_data_pipeline_dag' AIRFLOW_CTX_TASK_ID='run_bronze_load' AIRFLOW_CTX_EXECUTION_DATE='2025-09-16T15:05:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-09-16T15:05:00+00:00'
[2025-09-16T15:24:18.897+0000] {bronze_load.py:30} INFO - Loaded 118647 rows from Parquet: /opt/airflow/scripts/opensky/../../data/kafka_parquet/opensky_batch_20250916_152417.parquet
[2025-09-16T15:24:18.897+0000] {bronze_load.py:31} INFO - Beginning to extract and load data into bronze_info table.
[2025-09-16T15:24:18.905+0000] {base.py:83} INFO - Using connection ID 'postgres_default' for task execution.
[2025-09-16T15:24:18.932+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etl/bronze/opensky/bronze_load.py", line 43, in insert_into_bronze_ddl
    clean_row = sanitize_row(full_row)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etl/bronze/opensky/bronze_load.py", line 15, in sanitize_row
    return [None if (isinstance(x, float) and (math.isnan(x) or math.isinf(x))) else x for x in row]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/etl/bronze/opensky/bronze_load.py", line 15, in <listcomp>
    return [None if (isinstance(x, float) and (math.isnan(x) or math.isinf(x))) else x for x in row]
                                               ^^^^
NameError: name 'math' is not defined
[2025-09-16T15:24:18.937+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=opensky_data_pipeline_dag, task_id=run_bronze_load, execution_date=20250916T150500, start_date=20250916T152418, end_date=20250916T152418
[2025-09-16T15:24:18.940+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 102 for task run_bronze_load (name 'math' is not defined; 9302)
[2025-09-16T15:24:18.950+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-09-16T15:24:18.961+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
